akka-actor/src/main/scala/akka/pattern/package.scala

```diff
@@ -40,6 +40,51 @@ import akka.actor._
  * ask(actor, message);
  * }}}
  */
-package object pattern extends PipeToSupport with AskSupport with GracefulStopSupport with FutureTimeoutSupport {
+package object pattern
+  extends PipeToSupport
+  with AskSupport
+  with GracefulStopSupport
+  with FutureTimeoutSupport
 
+package pattern {
+  /**
+   * == Extended Versions Of Akka Patterns ==
+   *
+   * This subpackage contains extended versions of Akka patterns.
+   *
+   * Currently supported are:
+   *
+   * <ul>
+   * <li><b>ask:</b> create a temporary one-off actor for receiving a reply to a
+   * message and complete a [[scala.concurrent.Future]] with it; returns said
+   * Future.</li>
+   * a message.</li>
+   * </ul>
+   *
+   * In Scala the recommended usage is to import the pattern from the package
+   * object:
+   * {{{
+   * import akka.pattern.extended.ask
+   *
+   * ask(actor, askSender => Request(askSender)) // use it directly
+   * actor ask (Request(_))   // use it by implicit conversion
+   * }}}
+   *
+   * For Java the patterns are available as static methods of the [[akka.pattern.Patterns]]
+   * class:
+   * {{{
+   * import static akka.pattern.Patterns.ask;
+   *
+   * // for Akka compiled with Scala 2.12
+   * ask(actor, askSender -> new Request(askSender));
+   *
+   * // for Akka compiled with Scala 2.11
+   * ask(actor, new akka.japi.Function<ActorRef, Object> {
+   *   Object apply(ActorRef askSender) {
+   *     return new Request(askSender);
+   *   }
+   * });
+   * }}}
+   */
+  package object extended extends ExplicitAskSupport
```

Comment:
This should not be a package object and it should also not be within the package object for `akka.pattern`—package objects are strange beasts. I think this should be a normal `object` within the `akka.pattern` package.